File: proxy/tests/script_engine/body_sub.js
/// <reference path="../../script_libs/js/roxy.d.ts" />
/** @type {Extension} */
const bodySub = {
  request(flow) {
    flow.request.body.text = flow.request.body.text.replace("replaceme", "gone");
  },
  response(flow) {
    flow.response.body.text = flow.response.body.text.replace("to_go", "it_went");
  }
};
globalThis.extensions = [bodySub];

File: proxy/tests/script_engine/body_sub.py
from roxy import Extension


class SubBody(Extension):
    def request(self, flow):
        flow.request.body.text = flow.request.body.text.replace("replaceme", "gone")

    def response(self, flow):
        flow.response.body.text = flow.response.body.text.replace("to_go", "it_went")


Extensions = [SubBody()]
